server/lib/pbench/indexer.py

```diff
@@ -3524,36 +3979,80 @@ def __init__(self, idxctx, tbarg, tmpdir, extracted_root):
         # It's what ties all of the relevant documents together.
         self.run_metadata["id"] = md5sum
 
+    def gen_files_by_partial_path(self, path):
+        """Generator for all files in the tar ball which match the given path
+        pattern.
+        """
+        for member in self.members:
+            if member.isfile() and member.name.find(path) >= 0:
+                yield member.name
+
+    _iter_num_pat = re.compile(r"(?P<num>^[1-9][0-9]*)-")
+
     def get_iterations(self):
+        """Get the list of Iteration objects for this pbench tar ball.
+
+        Note: (in v0.67 and earlier)
+
+            * pbench-user-benchmark recorded the list of iterations as a set
+              of numbers, not on-disk directory names.
+
+            * pbench-specjbb2005 recorded the iterations in its
+              "iterations.txt" file instead of ".iterations" which means we
+              didn't have iterations properly recorded in the metadata.log
+              file; AND they were recorded as a set of numbers, not on-disk
+              directory names.
+
+        Some benchmarks (NOT including pbench-uperf, -fio, -trafficgen, or
+        -user-benchmark) do not record the iteration number separately from
+        the on-disk name.
+        """
         try:
             # N.B. Comma-separated list
             iterations_str = self.run_metadata["iterations"]
         except Exception:
-            # TBD - trawl through tb with some heuristics
+            # Since we don't have the iterations list in the metadata, we run
+            # through the tar ball members looking for directories that are
+            # most likely iterations.
             iterations = []
-            for path in self.tb.getnames():
-                path_els = path.split("/")
+            for member in self.members:
```

Comment:
pathlib would make life easier here as well.

Reply:
Here `pathlib` might be helpful, since we could do `path_els = Path(member.name).parts`.

Reply:
Let's track this in issue #1584 and make this a deliberate change in a separate commit.